src/main/riddl/LocationsEventsProjections/locationsEventsProjections.riddl

```diff
@@ -0,0 +1,71 @@
+  include "locationsEventsMessages"
+
+  projection LocationsEventsView is {
+     fields {
+        orgId: OrganizationId,
+        locationId: LocationId,
+        capacity: Integer,
+        reservationId: ReservationId,
+        reservationStart: TimeStamp,
+        reservationEnd: TimeStamp,
+        isPrivate: Boolean // we can use this for a sum of location, reservation, & venue privacy
+    }
+    handler LocationsEventsViewHandler is {
+        on event VenueEstablished {
+            then "find rows with venueId set to VenueEstablished.venueId and orgId set to VenueEstablished.info.orgId"
+            and "set rows' isPrivate to true if VenueEstablished.info.isPrivate is true"
+        }
+        on event VenueInfoUpdated {
+            then "find rows with venueId set to VenueInfoUpdated.venueId and orgId set to VenueEstablished.info.orgId"
+            and "set rows' isPrivate to true if VenueInfoUpdated.info.isPrivate is true"
+        }
+        on event VenueLocationsAdded {
+            then "find or create rows with venueId set to VenueLocationsAdded.venueId and locationIds set to VenueLocationsAdded.locationIds"
+        }
+        on event ReservationEstablished {
+            then "find or create rows with reservationId set to ReservationEstablished.reservationId"
+            and "set rows' reservationStart to ReservationEstablished.info.range.start"
+            and "set rows' reservationEnd to ReservationEstablished.info.range.end"
+            and "set rows' isPrivate to true if ReservationEstablished.info.isPrivate is true"
+        }
+        on event ReservationInfoUpdated {
+            then "find rows with reservationId set to ReservationInfoUpdated.reservationId"
+            and "set rows' reservationStart to ReservationInfoUpdated.info.range.start"
+            and "set rows' reservationEnd to ReservationInfoUpdated.info.range.end"
+            and "set rows' isPrivate to true if ReservationInfoUpdated.info.isPrivate is true"
+        }
+        on event LocationEstablished {
+            then "find or create rows with reservationId set to LocationEstablished.reservationIds"
+            and "set rows' capacity to LocationEstablished.locationInfo.capacity"
+            and "set rows' locationIds to LocationEstablished.locationIds"
+            and "set rows' isPrivate to true if LocationEstablished.locationInfo.isPrivate is true"
+        }
+        on event LocationReservationsAdded {
+            then "find or create rows with reservationId set to LocationReservationsAdded.reservationId"
+            and "set rows' locationId to LocationReservationsAdded.locationId"
+        }
+        on event LocationInfoUpdated {
+            then "find rows where locationId is LocationInfoUpdated.locationId"
+            and "set rows' capacity to LocationInfoUpdated.info.capacity"
+            and "set rows' isPrivate to true if LocationInfoUpdated.info.isPrivate is true"
+        }
+        on query ScheduleEasyEvent {
```

Comment:
A query should not be able to create a reservation or set attributes on a reservation.

Reply:
You're totally right, all creation happens from gateway after returning the potentially adjusted event request as a Success case